### ISSUE TITLE
Calendar_ TimeZone-Support

### ISSUE THREAD

**Issue by mbausche** [primefaces/primeng#13546]
### Describe the feature you would like to see added

Just to be sure before we start to implement a custom solution: There is still no built-in timeZone-Support in p-calender (we use verion 15). Correct?

**Comment by SoyDiego** [comment]
Hi @mbausche, I found this old post ( 3 years ago) of @cagataycivici about Calendar Timezone:
![image](https://github.com/primefaces/primeng/assets/19764334/93204330-242d-461c-bf10-b153a28b1058)

It's a very good idea to wait before starting something complex because, if I'm not mistaken, PrimeNG Team tries to implement that all their libraries have the same things. And if something is new in one, they have to talk about it as a team among all of them to see if they implement it in the others or something like that.
It is better to wait and see if someone from the PrimeNG Team gives you a better answer, but now they are also very focused on finishing everything that has to do with accessibility.

Thank you also for wanting to contribute and it would be great if timezone is implemented in the future.

**Comment by SoyDiego** [comment]
Hi again @mbausche, here you have a link to primelocale https://github.com/primefaces/primelocale

**Comment by blubcow** [comment]
It would definitely be nice to have a consistent timezone conversion set up.
For example, the user of a global business application is travelling around and sets his time zone to "GMT+0200".
**Problem? No!** Easy to do with Angular (At least to view the datetime).
Just use the pipe `{{ yourMeetingTime | date:'dd.MM.YY HH:mm' : 'GMT+0200' }}`

Next up: User wants to edit this value, "yourMeetingTime".
**With a PrimeNG calendar? Big problem.**
It would be easy to have it displayed in a specific timezone. But no.
Now we need **1)** to get the wanted time in GMT+0200. **2)** Then keep the values, but convert it to the users timezone (like Tokyo GMT+0900). **3)** Then create a Date object and pass it to PrimeNG. **4)** Get back the new Date. **5)** Convert the value back from GMT+0200 to GMT+0900 **6)** Create the Date object with the new value
Great fun!

Maybe just using the same functionality the Angular pipe is using, as an additional parameter, is a viable solution?
Then there would be no discussion about how the timezones work.
They would just display the same Date object, in another timezone. That's how I see the usage of Angular.
If a user selects one hours past the current display time in GMT+0200,
and his browser is set up to use Tokyo time GMT+0900,
then the returned date must be the Tokyo time plus one hour.

**This way two way binding would work flawlessly**, since both input and output are GMT+0900, but only the display and selection are in GMT+0200.
Or is it too complicated to implement?

I think in a globalized world this is an important feature, to let the user select which timezone he wants to view his application in.

**Comment by blubcow** [comment]
_If anyone is looking for it, here is a workaround with moment.js:_

You basically have to create a separate attribute, eg. `datePickerValue`
`this.datePickerValue = new Date(moment.tz(this.dateObject, "Europe/Vienna").format('YYYY-MM-DD HH:mm:ss'))`

Create a datepicker with a listener on change
`<p-calendar [ngModel]="datePickerValue" (ngModelChange)="onDatePickerUpdate($event)"></p-calendar>`

And convert it back to the browser timezone when the change is triggered
```
onDatePickerUpdate(newDatePickerValue: Date){
   this.dateObject = new Date(moment.tz(newDatePickerValue, "Europe/Vienna").format('YYYY-MM-DD HH:mm:ss'))
}
```

**Comment by blubcow** [comment]
```
import moment from "moment";

/**
 * Takes a date, and converts (only) the values to the selected timezone,.
 */
export function convertDateValuesToTimezone(date:Date, timeZone:string = "Europe/Vienna"): Date{
    return new Date(moment.tz(date, timeZone).format('YYYY-MM-DD HH:mm:ss'));
}

/**
 * Takes a date, and converts (only) the values to the browser timezone from a selected timezone.
 */
export function convertDateValuesFromTimezone(date:Date, timeZone:string = "Europe/Vienna"): Date{
    const offset = moment(date).tz(timeZone).utcOffset();
    return moment(date).utcOffset(offset, true).toDate();
}
```
